### PR TITLE
Removed: AnnotationDriver

### PR DESCRIPTION
**Pull request by JanGalek**
from PR: https://github.com/Kdyby/Doctrine/pull/304#issuecomment-353424426

### CODE REVIEW

**Comment by enumag** [comment]
Probably this [directory](https://github.com/Kdyby/Doctrine/tree/master/tests/KdybyTests/Doctrine/models/AnnotationDriver) should be removed as well.

Also the tests are failing.

**Comment by JanGalek** [comment]
test started to fail along with v3.3 create - https://travis-ci.org/Kdyby/Doctrine/builds/319848204

I'm looking to directory, I dont deleted because - https://github.com/Kdyby/Doctrine/blob/v3.3/tests/KdybyTests/Doctrine/config/entity-provider-merging.neon#L5

**Comment by enumag** [comment]
Also this code is probably no longer needed: https://github.com/Kdyby/Doctrine/blob/master/src/Kdyby/Doctrine/DI/OrmExtension.php#L612-L617

And maybe this can be removed too https://github.com/Kdyby/Doctrine/blob/master/src/Kdyby/Doctrine/DI/OrmExtension.php#L586-L588.

**Comment by enumag** [comment]
I added some commits to v3.3, rebase please.

**Comment by JanGalek** [comment]
I changed https://github.com/Kdyby/Doctrine/blob/master/src/Kdyby/Doctrine/DI/OrmExtension.php#L612-L617 because Annotation need reader.

Some test will must be update, native implementation load all entities in path. Example: ConvertMappingCommand method testDefaultCo

**Comment by JanGalek** [comment]
Test failing on

`PHP 7.1 --prefer-stable`
`PHP 7.2 --prefer-lowest --prefer-stable`
`PHP 7.2 --prefer-stable PHPSTAN=1`

other needed are ok.

**Comment by JanGalek** [comment]
@enumag ok, i added hotfix for test php <= 7.1,  but need better solution

File: src/Kdyby/Doctrine/QueryObject.php [hunk 1]
```diff
@@ -51,9 +51,10 @@
  *
  * @author Filip Procházka <filip@prochazka.su>
  */
-abstract class QueryObject extends Nette\Object implements Kdyby\Persistence\Query
+abstract class QueryObject implements Kdyby\Persistence\Query
 {
 
+	use Nette\SmartObject;
```
Comment (enumag) [inline src/Kdyby/Doctrine/QueryObject.php]: Add blank line after this one.

File: tests/KdybyTests/Doctrine/Console/GenerateEntitiesCommand.phpt [hunk 2]
```diff
@@ -34,12 +34,15 @@ class GenerateEntitiesCommandTest extends CommandTestCase
 		]);
 
 		$output = $applicationTester->getDisplay();
+		$output = str_replace('  ', '', $output);
+		$output = preg_replace( "/\r|\n/", '', $output );
```
Comment (enumag) [inline tests/KdybyTests/Doctrine/Console/GenerateEntitiesCommand.phpt]: `$output = preg_replace("/\r|\n/", '', $output);`

File: tests/KdybyTests/Doctrine/Console/GenerateEntitiesCommand.phpt [hunk 3]
```diff
@@ -34,12 +34,15 @@ class GenerateEntitiesCommandTest extends CommandTestCase
...
 
 		foreach (self::$entities as $entity) {
 			Assert::contains("Processing entity \"{$entity}\"", $output);
 		}
 		Assert::notContains(sprintf('Processing entity "%s"', \KdybyTests\Doctrine\Models2\Foo::class), $output);
-		Assert::contains('Entity classes generated to "' . realpath($destDir) . '"', $output);
+		Assert::contains('Entity classes generated to', $output);
+		Assert::contains(realpath($destDir), $output);
```
Comment (enumag) [inline tests/KdybyTests/Doctrine/Console/GenerateEntitiesCommand.phpt]: What is different about the output now that makes the old assert invalid?

**Comment by enumag** [comment]
> remove dir: models/AnnotationDriver - Entities are used in tests, have I change it and remove it ?

Which tests are they used in? I thought they were only used in the AnnotationDriver test which you removed.

File: tests/KdybyTests/Doctrine/Console/GenerateEntitiesCommand.phpt (hunk 3 above)
Comment (JanGalek) [reply tests/KdybyTests/Doctrine/Console/GenerateEntitiesCommand.phpt]: it was before fix output, I rollbacked it ;)

**Comment by JanGalek** [comment]
its used in https://github.com/Kdyby/Doctrine/blob/v3.3/tests/KdybyTests/Doctrine/Console/CommandTestCase.php#L39

nexts, in ExtensionTest

**Comment by enumag** [comment]
@JanGalek It's probably safe to remove the AnnotationDriver entities from both places you mentioned. Also fix the regexp lines pls.

**Comment by JanGalek** [comment]
@enumag ok, I found problem with Nette/Object, for test are used old version of kdyby/console, where its used, for example: https://github.com/Kdyby/Console/blob/v2.7.0/src/CliResponse.php

https://travis-ci.org/Kdyby/Doctrine/jobs/325823317#L501

**Comment by enumag** [comment]
That explains why it fails on `--prefer-lowest`, you can solve that by changing the constraint in composer.json (most likely you need to change constraint for not just console but for all nette and kdyby packages).

However it does not explain why [this build](https://travis-ci.org/Kdyby/Doctrine/jobs/327133060#L529) failed.

**Comment by JanGalek** [comment]
Now tests are ok, but phpstan has error.

**Comment by enumag** [comment]
The PHPStan failure is pretty clear and should be easy to fix:

```
Fatal error: Declaration of KdybyTests\DoctrineMocks\StatementMock::fetch($fetchStyle = NULL) must be compatible with Doctrine\DBAL\Driver\ResultStatement::fetch($fetchMode = NULL, $cursorOrientation = PDO::FETCH_ORI_NEXT, $cursorOffset = 0) in /home/travis/build/Kdyby/Doctrine/tests/KdybyTests/DoctrineMocks/StatementMock.php on line 22
```

**Comment by JanGalek** [comment]
@enumag Yes I chanched StatementMock, now nexts errors :)

**Comment by enumag** [comment]
These might be more tricky, possibly even bugs in PHPStan itself. First make sure that we have the latest PHPStan.

**Comment by JanGalek** [comment]
@enumag I updated phpstan:
`PHPStan - PHP Static Analysis Tool 0.9.1.0` so and now 100 errors

**Comment by enumag** [comment]
@JanGalek uups... Ok, revert PHPStan update for now, I'll merge this PR and someone can solve PHPStan errors in another PR. It can be you of course if you wish but it should be a separate PR in my opinion. Do you agree?

**Comment by JanGalek** [comment]
@enumag Yes, I agree

**Comment by enumag** [comment]
@JanGalek Merged. Thank you very much!

**Pull request merged**